monitorcontroller/monitorcontroller.go

```diff
@@ -98,16 +106,81 @@ func (monitor *MonitorController) Close() {
 	}
 }
 
-// SendMonitoringData sends monitoring data.
-func (monitor *MonitorController) SendMonitoringData(monitoringData cloudprotocol.NodeMonitoringData) {
+// SendNodeMonitoring sends monitoring data.
+func (monitor *MonitorController) SendNodeMonitoring(nodeMonitoring aostypes.NodeMonitoring) {
 	monitor.Lock()
-	defer monitor.Unlock()
```

Comment:
Don't see any reason to send event out the look.  lock, defer unlock is always preferable.

Reply:
monitor is unlocked right before sending the messages, because sendMessages locks monitor as well

Reply:
You use the channel here it is asynchronous.

Reply:
channel is async till the moment it is overflows, after that execution blocks on channel write. 
And if monitor is still locked it blocks execution of the sendMessages function. So it is required to unlock before sending the messages.

Reply:
SendOfflineMessages unit test demostraits that